From 157c516518268fded78680995e0459023bcdbc03 Mon Sep 17 00:00:00 2001
From: riddle_hsu <riddle_hsu@htc.com>
Date: Mon, 8 Jun 2015 16:00:26 +0800
Subject: [PATCH] Fix type casting for broadcast delay with service

There is only type "ResolveInfo" or "BroadcastFilter" will be put in
BroadcastRecord.receivers.
To fulfill the concept:
"Improved broadcast delays to not apply the delay if the next receiver
of the broadcast is going to run in the same process as the last one."

diff --git a/services/core/java/com/android/server/am/BroadcastQueue.java b/services/core/java/com/android/server/am/BroadcastQueue.java
index a279290..e0a13e4 100644
--- a/services/core/java/com/android/server/am/BroadcastQueue.java
+++ b/services/core/java/com/android/server/am/BroadcastQueue.java
@@ -438,7 +438,8 @@ public boolean finishReceiverLocked(BroadcastRecord r, int resultCode,
             ActivityInfo nextReceiver;
             if (r.nextReceiver < r.receivers.size()) {
                 Object obj = r.receivers.get(r.nextReceiver);
-                nextReceiver = (obj instanceof ActivityInfo) ? (ActivityInfo)obj : null;
+                nextReceiver = (obj instanceof ResolveInfo) ?
+                        ((ResolveInfo) obj).activityInfo : null;
             } else {
                 nextReceiver = null;
             }
